include/async/when_any.hpp

```diff
@@ -313,6 +313,8 @@ struct op_state<StopPolicy, Rcvr> {
     };
 
     constexpr auto start() & -> void {
+        static_assert(not unstoppable_token<stop_token_of_t<env_of_t<Rcvr>>>,
+                      "Starting when_any<> but it cannot stop!");
```

Comment:
suggestion: Would be nice if the static assert message explained why it cannot stop (because empty when_all never completes on its own _and_ the connected receiver does not allow for cancellation)